godot-kotlin/src/nativeCore/kotlin/godot/core/CoreType.kt

```diff
@@ -5,13 +5,13 @@ import kotlinx.cinterop.CValue
 import kotlin.native.concurrent.AtomicReference
 import kotlin.native.concurrent.freeze
 
-abstract class CoreType<T: CStructVar>(value: CValue<T>) {
-  private val ref = AtomicReference(value.freeze())
-  internal var _value: CValue<T>
-    get() = ref.value
-    set(value) {
-      ref.compareAndSet(ref.value, value.freeze())
-    }
+abstract class CoreType<T: CStructVar>(internal var _value: CValue<T>) {
```

Comment:
`value.freeze()` is quite expensive, specially doing 100 times per frame. We do lose thread-safety on core classes, but access to them will be in the main thread.